src/main/java/bio/terra/stairway/DbFunction.java

```diff
@@ -0,0 +1,14 @@
+package bio.terra.stairway;
+
+import bio.terra.stairway.exception.DatabaseOperationException;
+import java.sql.SQLException;
+
+/**
+ * We provide a custom functional interface to set the right thrown exceptions
+ *
+ * @param <R> The return value of the database function. If there is no return value, specify Void.
+ */
+@FunctionalInterface
+public interface DbFunction<R> {
```

Comment:
Since this is only used in the class `DbRetry` I don't think there's a benefit to declaring in a top level scope. Declaring it local to `DbRetry` would have the same effect and not pollute the global namespace.